feat: improve ui/ux learn map

moT01 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->
ToDo:

~~Add icons~~
~~ Add descriptions on superblocks~~
~~ Add intros in each block~~
 ~~Add breadcrumbs~~
~~Auto expand block to page they came from~~
~~Auto scroll to block~~
~~ Remove path to intro for hotkeys stuff~~
Probably remove block intro component and pages?
~~center misaligned buttons~~ 
~~Add map at the bottom~~
 ~~Don’t allow project expansion, just list the projects~~
~~Add Cert at the bottom with icon or checkmark after it gets claimed~~
~~Add Sign in button if not signed in~~
~~Fix inevitably broken tests~~

Here's a table of the behavior I implemented when someone arrives on one of these map pages. The place they are coming from is on the left - the top row is the state of the app, the block to expand and the block to auto scroll to

|                                       |unAuthed - Expand|unAuthed - Scroll to|SignedIn - Expand|SignedIn - Scroll to|
|------------------------|-------------------|---------------------|------------------|-------------------|
|breadcrumb block click | clicked block         | clicked block           | clicked block       | clicked block         |
|anywhere else                | first block              | top of page              | currentChallenge block or first block | currentChallenge block or first block

moT01 committed:
feat: improve ui/ux learn map

scissorsneedfoodtoo committed:
feat: basic superblock intro text working

scissorsneedfoodtoo committed:
fix: intro schema

scissorsneedfoodtoo committed:
feat: first drafts of superblock intros and images

moT01 committed:
feat: add breadcrumbs to challenges

moT01 committed:
feat: add map to superBlock intros

moT01 committed:
feat: list project blocks

moT01 committed:
feat: add certs to super block pages

scissorsneedfoodtoo committed:
feat: basic block descriptions

scissorsneedfoodtoo committed:
fix: intro files and schema

moT01 committed:
feat: adjust project list icons

moT01 committed:
fix: ocd

moT01 committed:
fix: revert take home projects + simplify code + tidy css

scissorsneedfoodtoo committed:
fix: refactor intro.json for easier localization

scissorsneedfoodtoo committed:
feat: update Block.js to use localized block and superblock titles

scissorsneedfoodtoo committed:
fix: update SuperBlockIntro.js to use localized superblock titles

scissorsneedfoodtoo committed:
fix: intro.json typo

scissorsneedfoodtoo committed:
fix: undefined keys for project blocks

scissorsneedfoodtoo committed:
feat: allow localization for landing and learn maps

moT01 committed:
fix: rebase + fix conflicts + refactor schema to match new structure + add function for new intro.json files

moT01 committed:
fix: uncomment functions for intro validation + fix intro files to pass tests

moT01 committed:
feat: move block component next to its parent

moT01 committed:
feat: add projects block to all superBlocks + add missing JS block

moT01 committed:
fix: move styles from map.css to intro.css

scissorsneedfoodtoo committed:
fix: cleanup and wording

scissorsneedfoodtoo committed:
fix: wording and spacing before other certs

scissorsneedfoodtoo committed:
feat: prevent intro pages from rendering

scissorsneedfoodtoo committed:
fix: move block descriptions out of dropdown, add styles

scissorsneedfoodtoo committed:
fix: localize more SuperBlockIntro text

scissorsneedfoodtoo committed:
fix: styles after rebase

scissorsneedfoodtoo committed:
feat: localize more SuperBlockIntro text

scissorsneedfoodtoo committed:
fix: update intro.json files and intro schema

scissorsneedfoodtoo committed:
feat: add js ds and algos text

scissorsneedfoodtoo committed:
feat: render intro text dangerously for code highlighting

moT01 committed:
feat: add expanding and scrolling to blocks

moT01 committed:
fix: simplify component exports

moT01 commented:
This will be removing the introduction challenges from the map and replacing them with a small description. Do we want to just completely get rid of the block intro pages? And I suppose the files as well? I think I would say yes.

moT01 committed:
fix: remove introPath from everywhere - muahahaha

scissorsneedfoodtoo commented:
@moT01, thanks for removing the intro paths. And awesome commit message, too haha.

I'm still using the intro pages to help write the new descriptions, but yeah, I'm all for removing those files if we can. I think I was able to make it so those pages won't build in a stashed commit somewhere, but I didn't experiment with removing those files / directories yet.

scissorsneedfoodtoo committed:
feat: add front end libraries text

scissorsneedfoodtoo committed:
feat: add data visualization text

scissorsneedfoodtoo committed:
feat: add apis and microservices text

scissorsneedfoodtoo committed:
feat: add quality assurance text

scissorsneedfoodtoo committed:
fix: npm capitalization

moT01 committed:
fix: bold block titles + minor clean up

moT01 committed:
fix: cypress tests (hopefully)

moT01 committed:
feat: add coding interview block intros

moT01 committed:
feat: add the rest of the intro descriptions

scissorsneedfoodtoo committed:
fix: minor tweaks and additions to intro text

scissorsneedfoodtoo committed:
fix: sync intro.json files

scissorsneedfoodtoo committed:
fix: update intro-schema.js

scissorsneedfoodtoo committed:
feat: add icons to learn and landing map

moT01 committed:
feat: add placeholders for upcoming projects + default arr variable to array

moT01 committed:
feat: add alt text for big superBlock icons + update schema and other languages

moT01 committed:
fix: lighthouse accesiblity header problem

moT01 committed:
fix: lighthouse accessibility - no label around select element

moT01 committed:
fix: lighthouse accessibility - ul list can only contain li

moT01 committed:
fix: lighthouse SEO - tap targets appropriately sized

moT01 committed:
fix: map misalignment

moT01 committed:
feat: extract a few components out of main superBlock page

moT01 committed:
fix: extract challenges out of block component into its own

moT01 committed:
fix: update intro.json files

moT01 committed:
fix: minor css fixes

scissorsneedfoodtoo committed:
fix: align superblock, block, and button padding

scissorsneedfoodtoo committed:
fix: adjust spacers for blocks and superblock, move block headers up

scissorsneedfoodtoo committed:
fix: change dropdown tutorial text to h4, lower fontweight of text and count

scissorsneedfoodtoo committed:
fix: remove extra initializeExpandedState call

This gets rid of the "Cannot update during an existing state transition..." warning

scissorsneedfoodtoo committed:
fix: change .find() to .some() to pass boolean as prop

scissorsneedfoodtoo committed:
fix: simplify usage of spacing component

scissorsneedfoodtoo committed:
fix: add bottom margin to superblock intro images

scissorsneedfoodtoo committed:
feat: replace Block.test.js with Cypress test

scissorsneedfoodtoo committed:
fix: update jest snapshots

moT01 committed:
fix: hover effects on projects block and cert challenge

moT01 committed:
fix: combine title for i18n

moT01 committed:
fix: adjust spacers around sign in button

moT01 committed:
fix: adjust sign in button size

moT01 committed:
fix(lighthouse - perf): lazy load offscreen imgs + set explicit size + update snapshot

moT01 committed:
fix: split tutorials into two keys in json files

raisedadead commented:
I assume we will remove any extraneous files in a subsequent PR?
